crates/sui-faucet/src/metrics.rs

```diff
@@ -135,47 +135,79 @@ impl FaucetMetrics {
 
 impl MetricsCallbackProvider for RequestMetrics {
     fn on_request(&self, path: String) {
+        let normalized_path = normalize_path(&path);
+        if !is_path_tracked(normalized_path) {
+            return;
+        }
+
         self.total_requests_received
-            .with_label_values(&[path.as_str()])
+            .with_label_values(&[normalized_path])
             .inc();
     }
 
     fn on_response(&self, path: String, latency: Duration, _status: u16, grpc_status_code: Code) {
+        let normalized_path = normalize_path(&path);
+        if !is_path_tracked(normalized_path) {
+            return;
+        }
+
         self.process_latency
-            .with_label_values(&[path.as_str()])
+            .with_label_values(&[normalized_path])
             .observe(latency.as_secs_f64());
 
         match grpc_status_code {
             Code::Ok => {
                 self.total_requests_succeeded
-                    .with_label_values(&[path.as_str()])
+                    .with_label_values(&[normalized_path])
                     .inc();
             }
             Code::Unavailable | Code::ResourceExhausted => {
                 self.total_requests_shed
-                    .with_label_values(&[path.as_str()])
+                    .with_label_values(&[normalized_path])
                     .inc();
             }
             _ => {
                 self.total_requests_failed
-                    .with_label_values(&[path.as_str()])
+                    .with_label_values(&[normalized_path])
                     .inc();
             }
         }
     }
 
     fn on_start(&self, path: &str) {
+        let normalized_path = normalize_path(path);
+        if !is_path_tracked(normalized_path) {
+            return;
+        }
+
         self.current_requests_in_flight
-            .with_label_values(&[path])
+            .with_label_values(&[normalized_path])
             .inc();
     }
 
     fn on_drop(&self, path: &str) {
+        let normalized_path = normalize_path(path);
+        if !is_path_tracked(normalized_path) {
+            return;
+        }
+
         self.total_requests_disconnected
-            .with_label_values(&[path])
+            .with_label_values(&[normalized_path])
             .inc();
         self.current_requests_in_flight
-            .with_label_values(&[path])
+            .with_label_values(&[normalized_path])
             .dec();
     }
 }
+
+pub fn normalize_path(path: &str) -> &str {
+    if path.starts_with("/v1/status/") {
+        return "/v1/status";
+    }
+
+    path
+}
+
```

Comment:
Let's add some comments to explain why we're doing this. For reasons that I don't understand, it seems that in Grafana we see all sort of weird paths, so not sure why those are called?!?! Maybe it's bots that are scanning the deployment?

Reply:
Added!